packages/features/bookings/lib/get-booking.ts

```diff
@@ -161,3 +167,64 @@ export const getBookingByUidOrRescheduleUid = async (uid: string) => {
       : booking.attendees,
   };
 };
+
+/**
+ * Should only get booking attendees length for seated events
+ * @param uid
+ * @returns booking with masked attendee emails
+ */
+export const getBookingForSeatedEvent = async (uid: string) => {
+  const booking = await prisma.booking.findFirst({
+    where: {
+      uid,
+    },
+    select: {
+      id: true,
+      uid: true,
+      startTime: true,
+      attendees: {
+        select: {
+          id: true,
+        },
+      },
+      eventTypeId: true,
+      user: {
+        select: {
+          id: true,
+        },
+      },
+    },
+  });
+
+  if (!booking || booking.eventTypeId === null) return null;
+
+  // Validate booking event type has seats enabled
+  const eventType = await prisma.eventType.findFirst({
+    where: {
+      id: booking.eventTypeId,
+    },
+    select: {
+      seatsPerTimeSlot: true,
+    },
+  });
+  if (!eventType || eventType.seatsPerTimeSlot === null) return null;
+
+  const result: GetBookingType = {
+    ...booking,
+    // @NOTE: had to do this because Server side cant return [Object objects]
+    startTime: booking.startTime.toISOString() as unknown as Date,
+    description: null,
+    customInputs: null,
+    responses: {},
+    smsReminderNumber: null,
+    location: null,
+    // mask attendee emails for seated events
+    attendees: booking.attendees.map((attendee) => ({
+      ...attendee,
+      email: "",
+      name: "",
+      bookingSeat: null,
```

Comment:
Always hide current attendees

Reply:
Do we want to create a conditional where if `showAttendees` is true, then include the attendee name and emails? This is what we already show on the success page.